### ISSUE TITLE
Capture group generated doesn't work in Python 2.x

### ISSUE THREAD

**Issue by gnud** [firasdib/Regex101#1043]
## Question

If I type in substitution $0 or \0 when code is executed in python, the match is replaced with an unknown character.

Generated code:
```
subst = '$0'
result = re.sub(regex, subst, '', 0, re.MULTILINE)
```

If I use this code
```
subst = '\g<0>'
```
The match group is added, however \g is not in the Substitution Quick reference, but typing in down in the Substitution INPUT box works like $0 or \0, so what's the correct way now :) ?

**Comment by working-name** [comment]
Hi @gnud,

I'm sorry but I'm a bit confused as to what you're expecting to happen. 

Having `$0` or `\0` or `\g<0>` as part of the substitution string on the site does the same thing: print the full match in the replaced string. 

As far as the programming language is concerned, it depends whether `$0` or `\0` is used. As far as the unknown character ... maybe that's a remnant of your test string that wasn't matched or some odd characters escape sequence?

Check [this out as an example](https://regex101.com/r/6q2dHk/1).

**Comment by gnud** [comment]
Yes, but when I executed the code in Python 2.x, the $0 or \0 were replaced but unknown character. Without changing anything the generated code didn't work out of the box.
I was expecting if I write \0 and escaping is needed the generator to escape anything inside the variable subst.

As you can see here: https://repl.it/repls/SpottedMobileServers
The code is generate from your paste in Python generator section, I pasted the exact copy.

So the question is, why only the first \g<0> works, for some reason ?

**Comment by working-name** [comment]
I'm not sure why but `\0` or `\1` doesn't work in my tests with Python 2.7.14 although their [documentation](https://docs.python.org/2/library/re.html) clearly states it should work. I guess we'll have to stick to `\g<1>`? 

Maybe someone who is more python-abled will chime in :)

**Comment by gnud** [comment]
Well personally I can deal with it, but it with super if it's included in the Quick reference, so next time I forget the group \g<1> I or anyone else can see it in the list. Or correct me wrong if it's already present.

**Comment by working-name** [comment]
It's not yet present as a substitution string, but I don't see a problem with adding it. What do you think @firasdib?

**Comment by firasdib** [comment]
@working-name If it works im fine with it

**Comment by firasdib** [comment]
@working-name Should I add it, or have you got it in already?

**Comment by working-name** [comment]
Got it. Sent
